### PR TITLE
Add consolidated release notes for 2.7.0

### PR DESCRIPTION
**Pull request by zelinh**
### Description
Adding consolidated release notes for 2.7.0

### Issues Resolved
Part of #3230 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (seraphjiang) [inline release-notes/opensearch-release-notes-2.7.0.md:5]: @zelinh could we add highlight here?

OpenSearch Dashboards version 2.7 has introduced an improvement where administrators can now override the NodeJs version used to run the software. This modification has the potential to affect the launch logic and may require changes to existing infrastructure in order to properly launch OpenSearch Dashboards. Further information about this change can be found at the following link: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3508

Comment (bbarani) [reply release-notes/opensearch-release-notes-2.7.0.md:5]: @seraphjiang Just to clarify, by default there is no impact to normal OpenSearch users but adds additional capability for administrators to override NodeJs version used to run the software (when needed?). I see that there is a caveat stating that it has potential to affect the launch logic and may require changes to existing infrastructure in order to properly launch OpenSearch Dashboards when overriding it. Can we elaborate more on that rather than pointing to a PR?

Comment (bbarani) [reply release-notes/opensearch-release-notes-2.7.0.md:5]: Basically, point to a documentation / readme file with the information needed for launch logic when overriding the NodeJs version used to run the software.

Comment (seraphjiang) [reply release-notes/opensearch-release-notes-2.7.0.md:5]: The change is to allow user to choose NodeJS, the implementing has issue which is addressed in https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3955

For impacted customer, they has two options, modify their infrastructure code to adapt to the change, or apply above fix as workaround before we bring to release offically.

Comment (seraphjiang) [reply release-notes/opensearch-release-notes-2.7.0.md:5]: @bbarani if we could not put in highlight, i would suggest to add 3508 to break changes. 

This break change is not breaking functionality or api interface, but impact the launch behavior, although it was for another enhancement purpose. 

I'll leave this to team to make decision is better for the best of interest for community. We found this issue in early adoption and test of 2.7.

Comment (bbarani) [reply release-notes/opensearch-release-notes-2.7.0.md:5]: @jhmcintyre @seraphjiang Does it make sense to add a new "Known issues/bugs" section to release notes and list this there with the workaround?

Comment (jhmcintyre) [reply release-notes/opensearch-release-notes-2.7.0.md:5]: @bbarani @seraphjiang I would think adding a section called "Known issues" below the release highlights section would make more sense than categorizing this under "release highlights". (I don't know if I'd include "bugs" in that section title b/c this doesn't seem like a bug but open to feedback.

Comment (seraphjiang) [reply release-notes/opensearch-release-notes-2.7.0.md:5]: Thanks @bbarani  and @jhmcintyre the known issues works. 
This could help other customer to find workaround if they are facing same issue.

Comment (jhmcintyre) [reply release-notes/opensearch-release-notes-2.7.0.md:5]: @seraphjiang can you point me to where the workaround is documented and how the admin would apply it? I'm looking at 3508 and it's not entirely clear (at least to me).

Comment (seraphjiang) [reply release-notes/opensearch-release-notes-2.7.0.md:5]: The workaround is at https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3955/ not https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3508/

The instruction is to apply the same fix in 3955, and rebuild the OpenSearch Dashboards for the artifacts. For how to build , refer to https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md#building-artifacts

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (hdhalter) [inline release-notes/opensearch-release-notes-2.7.0.md:4]:
```suggestion
OpenSearch 2.7.0 includes a number of new and enhanced features for your search, observability, and analytics workloads. This release delivers production-ready functionality for segment replication, searchable snapshots, and the ability to add multiple data sources to a single dashboard — features that were previously released as experimental. Also included are new flat object fields for more efficient indexing, integrated observability features within OpenSearch Dashboards, new geospatial functionality, and updates designed to simplify administration of indexes and clusters.
```

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (hdhalter) [inline release-notes/opensearch-release-notes-2.7.0.md:31]: Can we add a link to the documentation release notes? https://github.com/opensearch-project/documentation-website/blob/main/release-notes/opensearch-documentation-release-notes-2.7.0.md

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (hdhalter) [inline release-notes/opensearch-release-notes-2.7.0.md:36]:
```suggestion
* Ml-commons introduced an API breaking change in 2.7 which changes the API response data. FE is updated accordingly. With this change, FE will no longer be compatible with OpenSearch version < 2.7 ([#154](https://github.com/opensearch-project/ml-commons-dashboards/pull/154))
```

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (hdhalter) [inline release-notes/opensearch-release-notes-2.7.0.md:42]:
```suggestion
* Implemented support for configuring up to 10 data filters for query and bucket-level monitors using the visual editor. ([#504](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/504))
```

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (hdhalter) [inline release-notes/opensearch-release-notes-2.7.0.md:43]:
```suggestion
* Implemented support for >, >=, <, and <= query operators for the doc-level monitor visual editor. ([#508](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/508))
```

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (hdhalter) [inline release-notes/opensearch-release-notes-2.7.0.md:67]:
```suggestion
* Error Prevention: Add notification, shrink, allocation, and rollup. ([#732](https://github.com/opensearch-project/index-management/pull/732))
```

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (hdhalter) [inline release-notes/opensearch-release-notes-2.7.0.md:115]:
```suggestion
* REST API for GET, PUT and DELETE ([#1502](https://github.com/opensearch-project/sql/pull/1502))
```

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (hdhalter) [inline release-notes/opensearch-release-notes-2.7.0.md:157]:
```suggestion
* Add exclude nodes setting ([#813](https://github.com/opensearch-project/ml-commons/pull/813))
```

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (hdhalter) [inline release-notes/opensearch-release-notes-2.7.0.md:158]:
```suggestion
* Set model state as partially loaded if unload model from partial nodes ([#806](https://github.com/opensearch-project/ml-commons/pull/806))
```

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (hdhalter) [inline release-notes/opensearch-release-notes-2.7.0.md:210]:
```suggestion
* Fixed a bug with doc-level trigger creation. ([#513](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/513))
```

**Review by hdhalter — COMMENTED** [review]
Did a quick read-through and made some suggestions.

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (jhmcintyre) [reply release-notes/opensearch-release-notes-2.7.0.md:5]: @seraphjiang @bbarani  taking a stab at writing this up. Is this accurate, and sufficiently comprehensive? 

Known Issues
 
Version 2.7.0 of OpenSearch Dashboards offers administrators the ability to manually override the version of Node.js used to run the software. If an administrator chooses this option, the change has the potential to affect the launch logic for OpenSearch Dashboards, in which case the administrator may need to modify their infrastructure code or apply a workaround to enable Dashboards to launch properly. The workaround involves making a change to the CLI shell scripts as described in this issue https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3955 and rebuilding the artifacts for Dashboards as described here https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md#building-artifacts. If the administrator does not choose to override the version of Node.js used to run the software, this will not be an issue.

Comment (seraphjiang) [reply release-notes/opensearch-release-notes-2.7.0.md:5]: Overall LGTM, except `If the administrator does not choose to override the version of Node.js used to run the software, this will not be an issue.` we might remove this as it will also impact cx not override the nodejs.

  thanks @jhmcintyre

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (jhmcintyre) [inline release-notes/opensearch-release-notes-2.7.0.md:14]: change to "dynamic tenant management"

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (jhmcintyre) [reply release-notes/opensearch-release-notes-2.7.0.md:31]: I'm good with this. @zelinh ?

**Review by jhmcintyre — CHANGES_REQUESTED** [review]

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (jhmcintyre) [reply release-notes/opensearch-release-notes-2.7.0.md:5]: Thanks @seraphjiang and I'm fine removing that last line. @bbarani, does this work for you?

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (zelinh) [reply release-notes/opensearch-release-notes-2.7.0.md:31]: @hdhalter @jhmcintyre Thanks for reviewing. I have added documentation release notes link below; and also updated based on reviews.

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (jhmcintyre) [inline release-notes/opensearch-release-notes-2.7.0.md:23]: should be "ML Commons" (with a capital L). And I think the section header can say "Change to ML Commons Dashboards" or "ML Commons Dashboards Update"

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (jhmcintyre) [inline release-notes/opensearch-release-notes-2.7.0.md:24]: Change "Ml-commons" to ML Commons. Also, what is FE? Should we spell it out each time it's used here?

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (jhmcintyre) [reply release-notes/opensearch-release-notes-2.7.0.md]: Please change "Ml" to "ML" in the subhead and copy, and spell out "ML Commons" (instead of Ml-commons) in the body copy.

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (dblock) [inline release-notes/opensearch-release-notes-2.7.0.md:24]: Period missing

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (dblock) [inline release-notes/opensearch-release-notes-2.7.0.md:60]: Period missing

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (dblock) [inline release-notes/opensearch-release-notes-2.7.0.md:196]: This is a fix, goes below

**Review by dblock — COMMENTED** [review]
Looks good. Some minor things I noticed, would be nice to get a mark down linter.

File: release-notes/opensearch-release-notes-2.7.0.md
Comment (zelinh) [reply release-notes/opensearch-release-notes-2.7.0.md]: Moved. Thanks!

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**
